api/v1alpha1/edgedevice_types.go

```diff
@@ -62,6 +65,14 @@ type HardwareProfileConfiguration struct {
 	Scope string `json:"scope,omitempty"`
 }
 
+type StorageConfiguration struct {
```

Comment:
WDYT about nesting the s3-specific configuration to allow for adding other types in the future?
Like:
```yaml
storage:
  s3:
    bucket_host:  ...
    ...
  other_storage_type:
    ...
```

Reply:
It might be sftp server, or network drive for example.